src/Icicle/Avalanche/Statement/Simp.hs

```diff
@@ -210,11 +211,16 @@ forwardStmts a_fresh statements
       If x ss es
        -> subS e x $ \x' -> If x' ss es
 
-      ForeachInts n from to ss
+      While t n end ss
+       -> do end'  <- sub e end
+             let e' = Map.delete n e
+             return (e', While t n end' ss)
```

Comment:
I am not sure, but I do not think this needs to remove `n` from the map any more, since `n` is an accumulator, and `n` is not bound here anyway